Serde is now a non-optional dependency

cberkhoff opened the pull request:
Removed the `enable-serde` flag and made Serde a non-optional dependency. Removed the `cfg` attributes that depended on the feature since now the library is included by default.

### Testing
Both unit tests and `./pre-commit` pass.

closes private-attribution/ipa#588

akoshelev reviewed (approved):
lgtm, thanks! The code coverage issues are probably coming from us not using these structs at all. Would you be able to test which ones are needed by simply deleting them and compiling the code? We may be able to clean up even more things

cberkhoff commented:
I went over the classes that were flagged as part of the coverage check and they were either:
a) Not used anywhere in the code. I removed those classes in the last commits.
b) Tests where not exercising the classes for which I added serialization/deserialization. This issue is orthogonal to this PR and rather tackle that separately.

akoshelev reviewed (approved):
lgtm, will merge it tomorrow
